python/hail/expr/functions.py

```diff
@@ -1783,10 +1783,10 @@ def sqrt(x) -> Float64Expression:
 
 @typecheck(s=expr_str)
 def _is_valid_allele(s) -> BooleanExpression:
-    return s.matches(r'^([ACGT]+)|\\*$')
+    return s.matches(r'^([ACGT]+)|(\*)|(<[\w:-]+>)$')
```

Comment:
ugh. will fix.

Reply:
the spec says this is a valid allele:
```
[13:123457[C
```

Reply:
oh hell no.

```
]<ctg1>: 329]A
```

Reply:
I'd be in favor of leaving this as "unknown", because wtf.

Reply:
We should also allow M (at least in the reference field) because the reference has an M in it, for example:

```
$ grep M Homo_sapiens_assembly38.fasta
CACACACCAAACAMCCCACACAACACACACACACCACACCACACAAACACAAACACACCACATCATGAACACACACATCACACACACACCACACACCCCA
             ^
```

which means "A or C":

https://www.bioinformatics.org/sms/iupac.html

Reply:
We currently can't import this from VCF because HTSJDK rejects it

Reply:
Well, this should work when that is fixed.